orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/WaitTask.java

```diff
@@ -44,17 +41,14 @@ public WaitTask(Clock clock) {
   public @Nonnull TaskResult execute(@Nonnull Stage stage) {
     WaitStage.WaitStageContext context = stage.mapTo(WaitStage.WaitStageContext.class);
 
-    if (context.getWaitTime() == null) {
-      return TaskResult.SUCCEEDED;
-    }
-
     Instant now = clock.instant();
 
-    if (context.isSkipRemainingWait()) {
+    if (context.isSkipRemainingWait() || context.getWaitDuration().isZero()) {
```

Comment:
This is maybe a silly minor comment, but do we need to special case the zero wait time here?  Wouldn't the start time plus zero always be before now in the next block which would simplify the number of cases to consider?

Reply:
yeah, good call - thank you!

Reply:
Also realized the `context.getWaitDuration()` check for `null` on line 72 wasn't necessary and removed it